fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentRootObject.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.fcrepo4;
+
+import edu.unc.lib.dl.fedora.FedoraException;
+import edu.unc.lib.dl.fedora.ObjectTypeMismatchException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Cdr;
+
+/**
+ * A repository object which represents the root of the content tree. Can only
+ * contain AdminUnit objects as direct children.
+ * 
+ * @author bbpennel
+ *
+ */
+public class ContentRootObject extends ContentContainerObject {
+
+	public ContentRootObject(PID pid, Repository repository, RepositoryObjectDataLoader dataLoader) {
+		super(pid, repository, dataLoader);
+	}
+
+	@Override
+	public ContentContainerObject addMember(ContentObject member) throws ObjectTypeMismatchException {
+		// TODO should only allow admin units, but allow folders for the moment
```

Comment:
Another benefit is that if we have to shuffle tickets around we won't have to change the TODOs.  One negative is that it is slightly weird to link to code on an unmerged branch that will go away, I'm not 100% sure if the link will stay live